crypto/signatures_test.go

```diff
@@ -2,58 +2,171 @@ package crypto
 
 import (
 	"crypto/rand"
+	"errors"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
```

Comment:
I dislike both `asset` and `require`, writing out the full set of things via  t.Error seems fine to me. Also iffy on mock, espeically for smaller stuff writing out the full mock manually seems sufficient.

I'm finally understanding what all this is supposed to look like.